### PR TITLE
GDPR enhancements

### PR DESCRIPTION
**Pull request by maennis-ep**
Description:
This PR captures some bug fixes, refactoring, and increased functionality surrounding managing GDPR and data policy consent.  These enhancements include:
- The `X-Ep-Data-Policy-Segments` header value is now exposed as a configuration instead of being hard-coded in the component
- The GDPR consent form now waits for all calls to complete before redirecting.
- The GDPR consent form now uses a zoom instead of making a number of GET requests equal to the number of data policies
- There is a new component which lives in the Profile Page for now.  It shows a list of data policies available based on store and `X-EP-Data-Policy-Segments`.  It also allows users to consent or revoke consent to any individual policy at any time.

I also removed the duplicated all-fields-required value in the fr-FR.json locale file.

Linting:
- [x] No linting errors

Component Updates:
- [x] Component package requires update on npm [@elasticpath/store-components](https://www.npmjs.com/package/@elasticpath/store-components)

Tests:
- [ ] E2E tests (npm test run with `e2e`)
- [x] Manual tests
You will need to set up an EP Store with Data Policies enabled.  More information on configuring Data Policies in EP can be found here: https://documentation.elasticpath.com/commerce-manager/docs/configuration/commerce-configuration.html#data-policies
1. Enable GDPR in ep.config.json
2. Open the PWA in a private browser
3. You will be prompted by the GDPR consent form
4. Select the checkbox and click the ACCEPT button
5. All calls to Cortex will now complete before the page refreshes.  I have left the page refresh in, although I am not sure if it is necessary or if the modal just need to close.
6. Authenticate as a REGISTERED user, or create a new registered user.
7. Navigate to `/profile`
8. Scroll to the bottom of the page where the Data Polices section is displayed
9. Give and Revoke consent to data policies.

Documentation:
- [x] Requires documentation updates
GDPR.dataPolicySegments will need to be added to https://documentation.elasticpath.com/storefront-react/docs/setup.html#configuration-parameter-descriptions.

More information on data policy segments and the `X-Ep-Data-Policy-Segments` header can be found here: https://documentation.elasticpath.com/commerce/docs/cortex/frontend-dev/managing-data-policy.html

### CODE REVIEW

File: components/src/ProfileGDPR/MockHttpResponses/GET/threePolicies_mixed_response.json [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+              }
+            ],
+            "data-policy-consent": "true",
+            "policy-name": "Order Information",
+            "policy-reference-key": "DP12347"
+          }
+        ]
+      }
+    ]
+  }
```
Comment (shaunmaharaj) [inline components/src/ProfileGDPR/MockHttpResponses/GET/threePolicies_mixed_response.json]: please add new line char

File: components/src/ProfileGDPR/MockHttpResponses/GET/onePolicy_notConsented_response.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            ],
+            "data-policy-consent": "false",
+            "policy-name": "Marketing Contact Information",
+            "policy-reference-key": "DP12346"
+          }
+        ]
+      }
+    ]
+  }
+
```
Comment (shaunmaharaj) [inline components/src/ProfileGDPR/MockHttpResponses/GET/onePolicy_notConsented_response.json]: please add new line char

**Comment by shaunmaharaj** [comment]
Thank you for this @maennis-ep!

@yulia-dnistrian can you please take a look at this since you worked on the initial implementation of it. I've added you as a code reviewer.

File: components/src/ProfileGDPR/MockHttpResponses/GET/onePolicy_notConsented_response.json (hunk 2 above)
Comment (shaunmaharaj) [reply components/src/ProfileGDPR/MockHttpResponses/GET/onePolicy_notConsented_response.json]: Somehow git is still complaining haha
image

Comment (maennis-ep) [reply components/src/ProfileGDPR/MockHttpResponses/GET/onePolicy_notConsented_response.json]: It looks like it is complaining about all the json files I added.  A quick google has turned up nothing.  Want me to keep looking, or are you okay accepting this as git error?

**Comment by shaunmaharaj** [comment]
@maennis-ep can you please address the conflicts, then I'll merge this one 😄

**Comment by maennis-ep** [comment]
@shaunmaharaj all taken care of!

**Pull request merged**